File: 2724-convert-an-array-into-a-2d-array-with-conditions/convert-an-array-into-a-2d-array-with-conditions.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        frequencies = [0 for _ in range(201)]

        for num in nums:
            frequencies[num] += 1
        
        result = [[]]
        totalElem = len(nums)

        while totalElem:
            for i, frequency in enumerate(frequencies):
                if frequency:
                    result[-1].append(i)
                    frequencies[i] -= 1
                    totalElem -= 1
            result.append([])
        
        result.pop()

        return result
        
        

File: 2724-convert-an-array-into-a-2d-array-with-conditions/convert-an-array-into-a-2d-array-with-conditions.ts
function findMatrix(nums: number[]): number[][] {
        const frequencies: { [key: number]: number } = {};  // Store frequencies of elements

        let maxFrequency = 0;  // Track the maximum frequency
        nums.forEach(num => {  // Count the frequency of each number
            frequencies[num] = (frequencies[num] || 0) + 1;
            maxFrequency = Math.max(maxFrequency, frequencies[num]);  // Update max frequency
        });

        const result: number[][] = Array.from({ length: maxFrequency }, () => []);  // Create a 2D array with rows equal to max frequency

        Object.keys(frequencies).forEach(key => {  // Distribute each number into the 2D array
            const num = parseInt(key);
            for (let j = 0; j < frequencies[num]; j++) {  // For each occurrence of the number
                result[j].push(num);  // Append the number to the corresponding row
            }
        });

        return result;
    }